### PR TITLE
improve focus example

### PR DESCRIPTION
**Pull request by derdilla**
Part of #130459. Adds a test to the [second focus example](https://api.flutter.dev/flutter/widgets/Focus-class.html#widgets.Focus.2) and makes the function of the example more clear.
## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Review by AdeelChudaray — COMMENTED** [review]
# A first-level heading
## A second-level heading
### A third-level heading

**Review by AdeelChudaray — COMMENTED** [review]
#138131

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Thanks!

File: examples/api/test/widgets/focus_scope/focus.1_test.dart [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          (WidgetTester tester) async {
+    await tester.pumpWidget(const MaterialApp(
+      home: Scaffold(
+        body: example.FocusableText(
+          'Item 0',
+          autofocus: false,
+        ),
+      ),
+    ));
+    // Autofocus needs to check no other node in the [FocusScope] is focused and
```
Comment (justinmc) [inline examples/api/test/widgets/focus_scope/focus.1_test.dart:22]: Super nit for readability: "needs to check no" => "needs to check that no"

File: examples/api/test/widgets/focus_scope/focus.1_test.dart [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      of: focusableTextFinder2,
+      matching: find.byType(Focus),
+    )).autofocus, isTrue);
+    final Container container2 = tester.widget<Container>(find.descendant(
+      of: focusableTextFinder2,
+      matching: find.byType(Container),
+    ));
+    expect(container2.color, Colors.red);
+  });
+  testWidgets('builds list showcasing focus traversal',
```
Comment (justinmc) [inline examples/api/test/widgets/focus_scope/focus.1_test.dart:61]: Needs a newline here between the tests.

**Review by justinmc — APPROVED** [review]
LGTM with nits 👍 

Thanks for helping to test these examples!

**Comment by A111one** [comment]
Welcome

**Comment by NobodyForNothing** [comment]
Can someone rerun Google testing?

**Comment by gspencergoog** [comment]
> Can someone rerun Google testing?

Done, but it looks like there are merge conflicts anyhow.

**Pull request merged**